specutils/io/default_loaders/generic_ecsv_reader.py

```diff
@@ -6,21 +6,33 @@
 from astropy.table import Table
 from ..generic_spectrum_from_table import generic_spectrum_from_table
 
+
 def identify_ecsv(origin, *args, **kwargs):
-# check if file can be opened with this reader
-# args[0] = filename
+    """Check if it's an ECSV file."""
     return (isinstance(args[0], six.string_types) and
-    # check if file is .ecsv (would be nice to validate the header too)
-        os.path.splitext(args[0].lower())[1] == '.ecsv'
-        )
+            os.path.splitext(args[0].lower())[1] == '.ecsv')
+
 
 @data_loader("generic-ecsv", identifier=identify_ecsv, dtype=Spectrum1D)
 def generic_ecsv(file_name, **kwargs):
     """
     Read a spectrum from an ECSV file, using generic_spectrum_from_table_loader()
     to try to figure out which column is which.
+
```

Comment:
It's a bit counter-intuitive that this doesn't have a `column_mapping` (I had a completely different comment I deleted after seeing this...).  So I suggest the following to reflect the difference for this function
```suggestion
    The ECSV columns must have units, as `generic_spectrum_from_table_loader`
    depends on this to determine the meaning of the columns.  For manual 
    control over the column to spectrum mapping, use the ASCII loader.

```

Reply:
I think it's better just to allow a column mapper here?

Reply:
Allowing a column mapper is fine. But the functionality of having it try to figure out the columns is still useful. One just needs to be clear about the logic if this loader is doing both. Meaning that the column mapper over-rides the attempt to identify the columns based on their units. 

Whatever is done here -- it seems like it should be identical for IPAC and tabular FITS, which are both tabular data that have support for units. So if we are going to use generic_table_from_spectrum if no column mapper is provided here, we should do the same in those loaders.

Reply:
@hcferguson well put. That's actually what's been implemented, and why I think we should continue exactly what you describe in this loader as well. As you've highlighted: all such tabular-style loaders will attempted to use a column mapper if provided, but will fallback to generating the spectrum from the table in cases where it's not defined.